index.d.ts

```diff
@@ -1,45 +1,69 @@
-import { Connection, Channel } from 'amqplib/callback_api';
-import fastify, { FastifyPlugin } from 'fastify';
+import { Connection, Channel } from 'amqplib';
+import { FastifyPluginAsync } from 'fastify';
 
 declare namespace fastifyAmqp {
-  interface FastifyAmqpConnObject extends Connection {}
+  type FastifyAmqpConnObject = Connection;
 
-  interface FastifyAmqpChannelObject extends Channel {}
+  type FastifyAmqpChannelObject = Channel;
 
   interface FastifyAmqpOptions {
+    /**
+     * Full connection URL
+     */
+    url?: string;
```

Comment:
Are missing a test to connection by `url`

Reply:
You're right. Added in the last commit.